code/modules/mining/mine_items.dm

```diff
@@ -235,7 +235,7 @@
 	var/obj/item/stack/flag/F = locate() in get_turf(src)
 
 	var/turf/T = get_turf(src)
-	if(!T || !istype(T,/turf/simulated/floor/asteroid))
+	if(!T || !istype(T,/turf/simulated/mineral))
```

Comment:
Being in the wall requires admin jumping or a teleporter, it's an edge case that isn't worth checking for.